File: Backend/dgulaf/src/main/java/com/dgulaf/service/ItemImageService.java
package com.dgulaf.service;

import com.dgulaf.model.ItemImage;
import com.dgulaf.repository.ItemImageRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ItemImageService {
    private final ItemImageRepository itemImageRepository;

    public ItemImageService(ItemImageRepository itemImageRepository) {
        this.itemImageRepository = itemImageRepository;
    }

    // 이미지 저장
    public ItemImage saveImage(ItemImage itemImage) {
        return itemImageRepository.save(itemImage);  // 이미지를 DB에 저장
    }

    // 특정 아이템의 이미지 조회 (분실물/습득물 구분)
    public List<ItemImage> getImagesByItemId(int itemId, boolean isLost) {
        return itemImageRepository.findImagesByItemIdAndIsLost(itemId, isLost);  // 아이템 ID와 isLost로 이미지 조회
    }
}

File: Frontend/dgu_laf/lib/service/item_image_service.dart
import 'dart:convert';
import 'package:dgu_laf/model/item_image.dart';
import 'package:http/http.dart' as http;

class ItemImageService {
  static const String baseUrl = 'http://192.168.219.105:8080';

  static Future<ItemImage> addImage(ItemImage image) async {
    final response = await http.post(
      Uri.parse('$baseUrl/images'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(image.toJson()),
    );

    if (response.statusCode == 200) {
      return ItemImage.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to add image: ${response.statusCode}');
    }
  }

  static Future<List<ItemImage>> getImagesByItemId(
      int itemId, bool isLost) async {
    final response = await http.get(
      Uri.parse('$baseUrl/images/$itemId?isLost=$isLost'),
    );

    if (response.statusCode == 200) {
      final List<dynamic> data = jsonDecode(response.body);
      return data.map((e) => ItemImage.fromJson(e)).toList();
    } else {
      throw Exception('Failed to fetch images: ${response.statusCode}');
    }
  }
}
